soem/ethercateoe.c

```diff
@@ -94,45 +94,43 @@ int ecx_EOEsetIp(ecx_contextt *context, uint16 slave, uint8 port, eoe_param_t *
       EOE_HDR_LAST_FRAGMENT);
    EOEp->frameinfo2 = 0;
   
-   /* The EoE frame will include "empty" IP/DNS entries, makes wireshark happy.
-    * Specification say they are optional, TwinCAT include empty entries.
-    */
    if (ipparam->mac_set)
    {
       flags |= EOE_PARAM_MAC_INCLUDE;
       memcpy(&EOEp->data[data_offset], ipparam->mac.addr, EOE_ETHADDR_LENGTH);
+      data_offset += EOE_ETHADDR_LENGTH;
    }
-   data_offset += EOE_ETHADDR_LENGTH;
    if (ipparam->ip_set)
    {
       flags |= EOE_PARAM_IP_INCLUDE;
       EOE_ip_uint32_to_byte(&ipparam->ip, &EOEp->data[data_offset]);
+      data_offset += 4;
```

Comment:
Perhaps better to do (here and other places below):
data_offset += sizeof(uint32_t);